src/coreclr/jit/codegencommon.cpp

```diff
@@ -1956,6 +1963,95 @@ void CodeGen::genGenerateMachineCode()
     /* The code is now complete and final; it should not change after this. */
 }
 
+//----------------------------------------------------------------------
+// genUpdateLiveRangesForTruncatedIGs -- find any instruction groups that have
+//   had their count altered and updated live ranges that refer to those groups
+//
+void CodeGen::genUpdateLiveRangesForTruncatedIGs()
+{
+#ifdef DEBUG
+    if (compiler->verbose)
+    {
+        printf("*************** In genUpdateLiveRangesForTruncatedIGs()\n");
+    }
+#endif
+    bool updatedGroup = false;
+
+    if (compiler->lvaCount > 0)
+    {
+        for (insGroup* ig = GetEmitter()->emitIGlist; ig; ig = ig->igNext)
```

Comment:
Why do we have to iterate all the IGs? Won't iterating over the `emitJumpList` and getting `idjIG` good enough?

Reply:
I think when I tried that I still hit assertions. I'll have to do some more testing and get back to you on that one.

Reply:
I believe you didn't get chance to verify this, yes?

Reply:
No I haven't verified it, and implementing as requested will make it irrelevant because the entire operation of changing emitted groups disappears. I also can't iterate the list after it's been merged and it must be merged before this so I can't really test now.